src/sqlfluff/core/rules/base.py

```diff
@@ -129,9 +131,17 @@ def __init__(self, edit_type, anchor, edit=None):
         # the parsed structure.
         self.edit = copy.deepcopy(edit)
         if self.edit:
+            # Check that any edits don't have a position marker set.
+            # We should rely on realignment to make position markers.
             # Strip position markers of anything enriched, otherwise things can get blurry
             for seg in self.edit:
-                seg.pos_marker = seg.pos_marker.strip()
+                if seg.pos_marker:
+                    # Developer warning.
+                    rules_logger.debug(
+                        "Developer Note: Edit segment found with preset position marker. "
+                        "These should be unset and calculated later. %r"
```

Comment:
The log message contains `%r`, but you're not passing a parameter. Should you be passing `seg` as a parameter to `debug()`?

Reply:
Great question. I'm not sure. I'll remove the reference.